models/repo.go

```diff
@@ -1231,7 +1231,7 @@ func CountUserRepositories(userID int64, private bool) int64 {
 }
 
 // Repositories returns all repositories
-func Repositories(opts *SearchRepoOptions) (_ RepositoryList, err error) {
+func Repositories(opts *SearchRepoOptions) (_ RepositoryList, count int64, err error) {
```

Comment:
Why need to examine the contents of `opts.Searcher`? 

https://github.com/go-gitea/gitea/blob/master/models/issue_indexer.go#L119
https://github.com/go-gitea/gitea/blob/master/routers/admin/repos.go#L28